### PR TITLE
Added DataTable row details expanding collapsing animation

### PR DESCRIPTION
**Pull request by cTux**
- [ADD] Added tests for animation.
- [ADD] Added animation for expanding/collapsing table row details.

### CODE REVIEW

File: src/DataTable/DataTable.js [hunk 1]
```diff
@@ -153,9 +154,22 @@ class DataTable extends WixComponent {
 
     if (rowDetails) {
       const showDetails = !!this.state.selectedRows[rowNum];
+      const getRowRenderedDetails = (rowData, rowNum) => {
+        if (rowDetailsAnimation) {
+          return (<Animator show={showDetails} height>
```
Comment (alonbt) [inline src/DataTable/DataTable.js]: you can remove Animator from here

File: src/DataTable/DataTable.js [hunk 2]
```diff
@@ -153,9 +154,22 @@ class DataTable extends WixComponent {
...
+        if (showDetails) {
+          return rowDetails(rowData, rowNum);
+        }
+      };
+
       rowsToRender.push(
         <tr key={`${rowNum}_details`}>
-          <td data-hook={`${rowNum}_details`} className={classNames(s.details, showDetails ? s.active : '')} colSpan={this.props.columns.length}>{showDetails && rowDetails(rowData, rowNum)}</td>
+          <td data-hook={`${rowNum}_details`} className={classNames(s.details, showDetails ? s.active : '')} colSpan={this.props.columns.length}>
+            {getRowRenderedDetails(rowData, rowNum)}
```
Comment (alonbt) [inline src/DataTable/DataTable.js]: ...and add Animator it here... (wrap getRowRenderDetails method)

**Comment by alonbt** [comment]
@lbelinsk in general Animations looks good but there was a bug there, we walked through it to find a way to make it great without that bug and now I'm waiting for @cTux to add the changes we discussed into this PR

File: src/DataTable/DataTable.spec.js [hunk 3]
```diff
@@ -132,19 +132,32 @@ describe('Table', () => {
...
       driver.clickRow(0);
+
+      // After clicking content will appear at once
       expect(driver.getRowDetailsText(0)).toBe(defaultProps.data[0].a);
       driver.clickRow(0);
       expect(driver.hasRowDetails(0)).toBe(true);
-      expect(driver.getRowDetailsText(0)).toBe('');
+
+      // When we clicking second tome to collapse content will disappear after a while (based on animation speed)
+      // expect(driver.getRowDetailsText(0)).not.toBe('');
```
Comment (lbelinsk) [inline src/DataTable/DataTable.spec.js]: Why won't this line work? I left it commented out because the tests fails here

Comment (cTux) [reply src/DataTable/DataTable.spec.js]: Because of `{showDetails && getRowRenderedDetails(rowData, rowNum)}`. It dissappears at once.

Comment (cTux) [reply src/DataTable/DataTable.spec.js]: I will remove unnecessary files and fix the test. Thank you!

**Comment by cTux** [comment]
@lbelinsk, please, take a look. Test was fixed. Row details page was fixed. Removed unnecessary file. Row details now has animation by default without options.

**Comment by lbelinsk** [comment]
There are some diffs on eyes.it, so please approve or fix if needed.
Let me know when it is ready

**Pull request merged**